### ISSUE TITLE
Secret name not updateable

### ISSUE THREAD

**Issue by pdanysz** [rancher/rancher#8727]
**Rancher Versions:**
Server: 1.5.9 / stable

**Docker Version:**
Docker version 1.12.6, build 3a094bd/1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**
Bare Metal / Centos 7

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single Node, external DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
I found two bugs:
Bug.I. Action->Edit button thown error
---------------------------------------------------
1. Go to the Secrets Menu via "Infrastructure"
2. Select secret (check the checkbox)
3. Select from Button Menu -> Actions -> Edit 

Bug.II Cannot change name of secret entry
-----------------------------------------------
1. and 2. the same
3. click button on secret row on the right, and check EDIT
4. Change the name and click SAVE

**Results:** 
About Bug.I
------------------------------------------------
Error in JS is thrown in browser console:
```
controller.js:33 Uncaught TypeError: this.get(...)[e] is not a function
    at n.applyBulkAction (controller.js:33)
    at o.join (ember.prod.js:345)
    at Function.l.join (ember.prod.js:22919)
    at ember.prod.js:11055
    at Object.p [as flaggedInstrument] (ember.prod.js:19341)
    at l (ember.prod.js:11054)
    at n.executeBulkAction (sortable-table.js:64)
    at n.send (ember.prod.js:39308)
    at n [as send] (ember.prod.js:23950)
    at ember.prod.js:11517
```
And then nothing is happening. No popup.

About Bug.II
--------------------------------------------------
Popup closes but the name of the secret are not changed.
But changing the Description is working.

**Comment by tfiduccia** [comment]
Version - master 5/10
This is indeed broken with "controller.js:33 Uncaught TypeError: this.get(...)[e] is not a function" error.

**Comment by vincent99** [comment]
@ibuildthecloud name isn't marked updatable in API

**Comment by tfiduccia** [comment]
Version - 2.0 master 3/19
Verified fixed (this is updatable in the 2.0 product)
